### PR TITLE
Visitor can see side articles

### PR DESCRIPTION
**Pull request by sundqvistjohan**
[PT Link](https://www.pivotaltracker.com/story/show/170676811)

Adds index action to `Api::V1::ArticlesController` with pagination (will_paginate gem)
Adds index serializer with truncation of article `:body` to 75 chars
Request specs and serializer specs for the above

### CODE REVIEW

File: app/serializers/articles/index_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Articles::IndexSerializer < ActiveModel::Serializer
+  include ActionView::Helpers::TextHelper
+
+  attributes :id, :title, :body
+
+  def body
+    truncate(object.body, length: 75)
+  end
+
+end
```
Comment (tochman) [inline app/serializers/articles/index_serializer.rb:10]: I can't see any specs for this serializer

File: Gemfile [hunk 2]
```diff
@@ -9,6 +9,7 @@ gem 'pg', '>= 0.18', '< 2.0'
 gem 'puma', '~> 3.7'
 gem 'rack-cors', require: 'rack/cors'
 gem 'active_model_serializers', '~> 0.10.0'
+gem 'will_paginate', '~> 3.1.0'
```
Comment (tochman) [inline Gemfile:12]: I can't really see that you actually need this gem in this iteration of this functionality. The "problem" is that now that you have added it, I will need to see some tests for fetching the consequent pages. That is unnecessary complexity and a bit of premature optimization. I would have saved this for a later story if I ware you.

File: spec/requests/api/v1/articles_spec.rb [hunk 3]
```diff
@@ -1,32 +1,62 @@
-RSpec.describe 'GET /api/v1/articles/:id', type: :request do
+RSpec.describe Articles, type: :request do
```
Comment (tochman) [inline spec/requests/api/v1/articles_spec.rb]: The previous format was actually better. Also, you don't have a constant named Articles. If you really need to use this naming, why not wrap it in quotes?

Comment (tochman) [reply spec/requests/api/v1/articles_spec.rb]: But okay, I see what you are doing below... I would call this "Articles endpoint" or something like that.

File: app/serializers/articles/index_serializer.rb (hunk 1 above)
Comment (sundqvistjohan) [reply app/serializers/articles/index_serializer.rb:10]: Specs for Index serializer added.

File: Gemfile (hunk 2 above)
Comment (sundqvistjohan) [reply Gemfile:12]: As discussed we will proceed with pagination at this stage but we see your point. Metadata added to paginated response, with request specs.

File: spec/requests/api/v1/articles_spec.rb (hunk 3 above)
Comment (sundqvistjohan) [reply spec/requests/api/v1/articles_spec.rb]: Request specs for ArticlesController split into `index_specs.rb` and `show_specs.rb`.

File: app/controllers/api/v1/articles_controller.rb [hunk 4]
```diff
@@ -1,13 +1,18 @@
...
+
   def show
     article = Article.find(params[:id])
     render json: article, serializer: Articles::ShowSerializer
   end
 
   private
-  
+
   def not_found
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb]: I would now move this one to the application controller and change the error message to `Article not found`

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,15 @@
...
+
+  protected
+
+  def meta_attributes(resource)    
+    {
+      current_page: resource.current_page,
+      next_page: resource.next_page,
+      prev_page: resource.previous_page,
+      total_pages: resource.total_pages,
+      total_count: resource.total_entries
```
Comment (oliverochman) [inline app/controllers/application_controller.rb:15]: test total entries in the req spec

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (sundqvistjohan) [reply app/controllers/application_controller.rb:15]: Updated

File: app/controllers/api/v1/articles_controller.rb (hunk 4 above)
Comment (sundqvistjohan) [reply app/controllers/api/v1/articles_controller.rb]: Done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**